### PR TITLE
docs(faq): clarify the versioning strategy

### PR DESCRIPTION
**Pull request by Narretz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Docs

**What is the current behavior? (You can also link to an open issue here)**
Information is not super clear

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:
clarify info on:
- When do features and breaking changes appear
- Relationship with Semver
- Compatibility of core and optional modules

### CODE REVIEW

File: docs/content/misc/faq.ngdoc [hunk 1]
```diff
@@ -25,37 +25,30 @@ So it's definitely not a plugin or some other native browser extension.
...
-code might include a breaking change. When reviewing the commit we can then decide whether there really is
-a breaking change and if it is appropriate for the branch to which it is being merged. If so, then we
-require that the commit message contains an appropriate breaking change message.
-
-Additionally, when a commit lands in our master repository it is synced to Google where we test it against
-over 2000 applications using the test suites of these applications. This allows us to catch regressions
-quickly before a release. We've had a pretty good experience with this setup. Only bugs that affect features
-not used at Google or without sufficient test coverage, have a chance of making it through.
+improvement, a considerable simplification of the code, a measurable performance improvement, or a better
+developer experience (especially with regard to updating to Angular).
```
Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: _updating_ --> _upgrading_ (?)

File: docs/content/misc/faq.ngdoc [hunk 2]
```diff
@@ -65,6 +58,44 @@ is also deprecated but we only remove the feature once it's removed from jQuery
 jqLite and jQuery. One such example is the `bind` method, deprecated in favor of `on` but unlikely to be removed
 from jqLite any time soon.
 
+#### What is the version compatibility between AngularJS main and optional modules?
+
+AngularJS code is separated into a main module ("angular"), and a few different optional modules
+("angular-animate", "angular-route" etc) that are dependant on the main module.
+When a new AngularJS version is released, all modules are updated to the new version.
+This means that the main module and the optional modules must always have the exact same version,
+down the patch number, otherwise your application might break.
```
Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: _down the_ --> _down to the_ (?)

File: docs/content/misc/faq.ngdoc [hunk 3]
```diff
@@ -65,6 +58,44 @@ is also deprecated but we only remove the feature once it's removed from jQuery
...
+
+Therefore you must always explicitly lock down your dependencies, for example in the package.json,
+the following ensures that "angular" and "angular-animate" are always updated to the same version.
+
+```
+{
+  "angular": "~1.6.0",
+  "angular-animate": "~1.6.0"
+}
```
Comment (mgol) [inline docs/content/misc/faq.ngdoc]: Maybe add another example where versions are really locked as well?

**Review by mgol — CHANGES_REQUESTED** [review]

File: docs/content/misc/faq.ngdoc [hunk 4]
```diff
@@ -65,6 +58,44 @@ is also deprecated but we only remove the feature once it's removed from jQuery
...
+```
+
+#### How does AngularJS ensure code quality and guard against regressions?
+
+When adding new code to branches of AngularJS, have a very stringent commit policy:
```
Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: Not sure what "branches" refers to :grin: _Update: I get it now (after reading further below), but people won't know._
Also: _have_ --> _we have_ (?)

File: docs/content/misc/faq.ngdoc [hunk 5]
```diff
@@ -65,6 +58,44 @@ is also deprecated but we only remove the feature once it's removed from jQuery
...
+
+- Every commit must contain tests and documentation updates alongside the code changes and that all the
```
Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: _and that all_ --> _and all_ (?)

File: docs/content/misc/faq.ngdoc [hunk 6]
```diff
@@ -65,6 +58,44 @@ is also deprecated but we only remove the feature once it's removed from jQuery
...
+tests must pass;
+- Commit messages must be written in a specific manner that allows us to parse them and extract the changes
+for release notes.
```
Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: Let's add a link to the commit message guidelines (or CONTRIBUTING.md).

File: docs/content/misc/faq.ngdoc (hunk 3 above)
Comment (Narretz) [inline docs/content/misc/faq.ngdoc]: You mean locked down to the patch version without ~?

File: docs/content/misc/faq.ngdoc [hunk 7]
```diff
@@ -65,6 +58,44 @@ is also deprecated but we only remove the feature once it's removed from jQuery
...
+The AngularJS code base has a very large set of unit tests and end-to-end tests. This means that a breaking change will require one or more tests to be changed to allow the
+tests to pass. So when a commit includes tests that are being removed or modified, this is a flag that the
+code might include a breaking change. When reviewing the commit we can then decide whether there really is
+a breaking change and if it is appropriate for the branch to which it is being merged. If so, then we
+require that the commit message contains an appropriate breaking change message.
+
+Additionally, commits are periodically synced to Google where we test it against applications using
+the test suites of these applications. This allows us to catch regressions
+quickly before a release. We've had a pretty good experience with this setup. Only bugs that affect features
+not used at Google or without sufficient test coverage, have a chance of making it through.
```
Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: I am not sure the last two paragraphs are super-relevant to external contributors.

File: docs/content/misc/faq.ngdoc (hunk 3 above)
Comment (mgol) [inline docs/content/misc/faq.ngdoc]: Yes, to indicate that if you lock to a specific version, all versions need to be the same.

File: docs/content/misc/faq.ngdoc (hunk 7 above)
Comment (Narretz) [inline docs/content/misc/faq.ngdoc]: I assume they are meant to inspire confidence in the stability of AngularJS (I've just reworded them from their previous state)

Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: I didn't notice they were there before :confused: Fair enough.
👍 on rewording.

**Comment by Narretz** [comment]
Updated, PTAL

File: docs/content/misc/faq.ngdoc [hunk 8]
```diff
@@ -76,14 +76,24 @@ the following ensures that "angular" and "angular-animate" are always updated to
...
+}
+```
+
+
 #### How does AngularJS ensure code quality and guard against regressions?
 
-When adding new code to branches of AngularJS, have a very stringent commit policy:
+When adding new code to AngularJS, we have a very stringent commit policy:
 
 - Every commit must contain tests and documentation updates alongside the code changes and that all the
```
Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: _and that all_ --> _and all_ (or is it intentional?)

File: docs/content/misc/faq.ngdoc [hunk 9]
```diff
@@ -130,8 +140,8 @@ Yes. See instructions in {@link downloading}.
 ### What browsers does AngularJS work with?
 
 We run our extensive test suite against the following browsers: the latest versions of Chrome,
-Firefox, Safari, and Safari for iOs, as well as Internet Explorer versions 9-11. See {@link guide/ie
-Internet Explorer Compatibility} for more details on supporting legacy IE browsers.
+Firefox, Safari, and Safari for iOs, Edge, as well as Internet Explorer versions 9-11. See
```
Comment (gkalpak) [inline docs/content/misc/faq.ngdoc]: _iOs_ --> _iOS_

(Are we running against Edge yet?)

**Review by gkalpak — APPROVED** [review]
A couple of typos. LGTM otherwise.

File: docs/content/misc/faq.ngdoc (hunk 9 above)
Comment (mgol) [inline docs/content/misc/faq.ngdoc]: We're not: issue #13687, PR #14401.

Yes, we should fix that. :D

**Review by mgol — APPROVED** [review]

File: docs/content/misc/faq.ngdoc (hunk 9 above)
Comment (Narretz) [inline docs/content/misc/faq.ngdoc]: I removed it again. Edge had some problems with the script security checks last time i checked

**Comment by Narretz** [comment]
Thanks for the reviews!

**Pull request merged**